modules/ximgproc/src/edgepreserving_filter.cpp

```diff
@@ -0,0 +1,387 @@
+							    intensity.val[1] *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]);
+							weight[globalPosX]
+							      [globalPosY][2] +=
+							    intensity.val[2] *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]);
+
+							// compute final image
+							imageResult[globalPosX]
+							           [globalPosY]
+							           [0] +=
+							    intensity.val[0] *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    ArithmeticMean[0];
+							imageResult[globalPosX]
+							           [globalPosY]
+							           [1] +=
+							    intensity.val[1] *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    ArithmeticMean[1];
+							imageResult[globalPosX]
+							           [globalPosY]
+							           [2] +=
+							    intensity.val[2] *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    (threshold -
+							     pixelwiseDist
+							         [subPosX]
+							         [subPosY]) *
+							    ArithmeticMean[2];
+						};
+					};
+				};
+			};
+		};
+
+		// compute final image
+		for (int globalPosX = 0; globalPosX < (int)imageResult.size();
+		     globalPosX++)
+		{
+			for (int globalPosY = 0;
+			     globalPosY < (int)imageResult[globalPosX].size();
+			     globalPosY++)
+			{
+				// cout << "globalPosX: " << globalPosX << "/"
+				// << dst->cols << "," << imageResult.size () <<
+				// "\tglobalPosY: " << globalPosY << "/" <<
+				// dst->rows << "," <<imageResult.at
+				// (globalPosX).size () << endl;
+
+				// add image to result
+				cv::Vec3b intensity =
+				    src->at<cv::Vec3b>(globalPosY, globalPosX);
+				imageResult[globalPosX][globalPosY][0] +=
+				    (double)intensity.val[0];
+				imageResult[globalPosX][globalPosY][1] +=
+				    (double)intensity.val[1];
+				imageResult[globalPosX][globalPosY][2] +=
+				    (double)intensity.val[2];
+
+				// normalize using weight
+				imageResult[globalPosX][globalPosY][0] /=
+				    (weight[globalPosX][globalPosY][0] + 1);
+				imageResult[globalPosX][globalPosY][1] /=
+				    (weight[globalPosX][globalPosY][1] + 1);
+				imageResult[globalPosX][globalPosY][2] /=
+				    (weight[globalPosX][globalPosY][2] + 1);
+
+				// copy to output image frame
+				dst->at<cv::Vec3b>(globalPosY, globalPosX)[0] =
```

Comment:
Vec3b has `=`, `/=`, `+=`, etc. operators, so you could condense 3 lines into 1:

    dst->at<cv::Vec3b>(globalPosY, globalPosX) = imageResult[globalPosX][globalPosY];